### PR TITLE
adapt sdk to android 13 MAPSMBL-120

### PR DESCRIPTION
**Pull request by alexander-kulikovskii**
### Summary of changes

Soft-adoptating for API level 33. Right now we make our codebase compatible with new API level.

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [ ] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImpl.kt [hunk 1]
```diff
@@ -304,15 +304,15 @@ class CameraAnimationsPluginImpl : CameraAnimationsPlugin {
...
 
-      override fun onAnimationCancel(animation: Animator?) {
+      override fun onAnimationCancel(animation: Animator) {
         finishAnimation(animation, AnimationFinishStatus.CANCELED)
       }
 
-      override fun onAnimationRepeat(animation: Animator?) {}
+      override fun onAnimationRepeat(animation: Animator) {}
 
       private fun finishAnimation(animation: Animator?, finishStatus: AnimationFinishStatus) {
```
Comment (Chaoba) [inline plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImpl.kt:317]: The `?` in this method can also be removed

File: app/src/main/java/com/mapbox/maps/testapp/examples/terrain3D/SantaCatalinaActivity.kt [hunk 2]
```diff
@@ -41,8 +41,8 @@ class SantaCatalinaActivity : AppCompatActivity() {
   private var timeAnimator: TimeAnimator? = null
 
   override fun onCreate(savedInstanceState: Bundle?) {
-    super.onCreate(savedInstanceState)
     requestWindowFeature(Window.FEATURE_NO_TITLE)
+    super.onCreate(savedInstanceState)
```
Comment (tobrun) [inline app/src/main/java/com/mapbox/maps/testapp/examples/terrain3D/SantaCatalinaActivity.kt:45]: could you explain this change?

**Review by tobrun — APPROVED** [review]

File: app/src/main/java/com/mapbox/maps/testapp/examples/terrain3D/SantaCatalinaActivity.kt (hunk 2 above)
Comment (alexander-kulikovskii) [reply app/src/main/java/com/mapbox/maps/testapp/examples/terrain3D/SantaCatalinaActivity.kt:45]: robo-test found crash if the order leave without changes

Comment (alexander-kulikovskii) [reply app/src/main/java/com/mapbox/maps/testapp/examples/terrain3D/SantaCatalinaActivity.kt:45]:
```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.mapbox.maps.testapp/com.mapbox.maps.testapp.examples.terrain3D.SantaCatalinaActivity}: android.util.AndroidRuntimeException: requestFeature() must be called before adding content
```

**Pull request merged**